src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1460 @@
+        )
+
+
+@add_start_docstrings(
+    """Beit3ForImagesAndTextClassification with a MLP head on top of Beit3Model, for images and text classification tasks
+    such as NLVR2""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3ForImagesAndTextClassification(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super(Beit3ForImagesAndTextClassification, self).__init__(config)
+        self.beit3 = Beit3Model(config)
+        self.classifier = Beit3MLP(config)
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_VISUAL_REASONING_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=SequenceClassifierOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids,
+        pixel_values,
+        attention_mask,
+        output_hidden_states=None,
+        output_attentions=None,
+        return_dict=None,
+        labels=None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3ForImagesAndTextClassification
+        >>> from PIL import Image
+        >>> import requests
+        >>> import torch
+
+        >>> model = Beit3ForImagesAndTextClassification.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+
+        >>> right_image_url = "https://i.ytimg.com/vi/DtTND8frecg/hqdefault.jpg"
+        >>> right_image = Image.open(requests.get(right_image_url, stream=True).raw)
+        >>> left_image_url = "https://cevgroup.org/wp-content/uploads/2018/10/WAP-7_class_electric_locomotive_of_Indian_Railways.jpg"
+        >>> left_image = Image.open(requests.get(left_image_url, stream=True).raw)
+
+        >>> text = "Power lines can be seen above the train in the image on the right."
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_224_nlvr2")
+        >>> inputs = processor(text=text, images=[left_image, right_image], return_tensors="pt")
+        >>> outputs = model(
+        ...     input_ids=inputs["input_ids"],
+        ...     pixel_values=inputs["pixel_values"].unsqueeze(0),
+        ...     attention_mask=torch.ones(inputs["input_ids"].shape),
+        ... )
+        >>> predicted_class_idx = outputs.logits.argmax(-1).item()
+        >>> predicted_class = model.config.id2label[predicted_class_idx]
+        >>> print("Predicted class:", predicted_class)
+        Predicted class: False
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        batch_size = input_ids.size()[0]
+        image1_values, image2_values = pixel_values.split(1, dim=1)
+        image1_values = image1_values.squeeze(1)
+        image2_values = image2_values.squeeze(1)
+        vision_input = torch.cat((image1_values, image2_values), dim=0)
+        language_input = torch.cat((input_ids, input_ids), dim=0)
+        attention_mask = torch.cat((attention_mask, attention_mask), dim=0)
```

Comment:
I don't think this will work for batched inputs?

Reply:
I have tested for batched inputs  and it works. Following are the shapes .
pixel_values:
1 sample : (1,2,3,224,224)
2 sample : (2,2,3,224,224)

input_ids :
1 sample : (1,_)
2 sample : (2,_)

image1_values :
1 sample : (1,3,224,224)
2 sample : (1,3,224,224)

image2_values :
1 sample : (1,3,224,224)
2 sample : (1,3,224,224)

vision_input :
1 sample : (2,3,224,224)
2 sample : (4,3,224,224)

language_input :
1 sample : (2,_)
2 sample : (4,_)